zerver/views/message_fetch.py

```diff
@@ -281,13 +289,16 @@ def get_messages_backend(
     return json_success(request, data=ret)
 
 
-@has_request_variables
+@typed_endpoint
 def messages_in_narrow_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    msg_ids: List[int] = REQ(json_validator=check_list(check_int)),
-    narrow: OptionalNarrowListT = REQ(converter=narrow_parameter),
+    *,
+    msg_ids: Json[List[int]],
+    narrow: Json[List[NarrowParameter]],
 ) -> HttpResponse:
+    narrow_parameter_list: OptionalNarrowListT = [x.model_dump() for x in narrow]
```

Comment:
Similar here with `OptionalNarrowListT`. The refactoring might be done in the same commit, or separately in a commit following this change.